avroparquet/src/test/java/docs/javadsl/AvroParquetSinkTest.java

```diff
@@ -0,0 +1,118 @@
+
+//#init-writer
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+//#init-writer
+
+
+public class AvroParquetSinkTest {
+
+  private final Schema schema =
+      new Schema.Parser()
+          .parse(
+              "{\"type\":\"record\",\"name\":\"Document\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"body\",\"type\":\"string\"}]}");
+  private final Configuration conf = new Configuration();
+  private final List<GenericRecord> records = new ArrayList<>();
+  private ActorSystem system;
+  private Materializer materializer;
+  private String folder = "javaTestFolder";
+  private final String file = "./" + folder + "/test.parquet";
+
+  @Before
+  public void setup() {
+    system = ActorSystem.create();
+    materializer = ActorMaterializer.create(system);
+    conf.setBoolean(AvroReadSupport.AVRO_COMPATIBILITY, true);
+    records.add(new GenericRecordBuilder(schema).set("id", "1").set("body", "body11").build());
+    records.add(new GenericRecordBuilder(schema).set("id", "2").set("body", "body12").build());
+    records.add(new GenericRecordBuilder(schema).set("id", "3").set("body", "body13").build());
+  }
+
+  @Test
+  public void createNewParquetFile() throws InterruptedException, IOException {
+    //#init-writer
+    Configuration conf = new Configuration();
+    conf.setBoolean(AvroReadSupport.AVRO_COMPATIBILITY, true);
+    ParquetWriter<GenericRecord> writer =
+        AvroParquetWriter.<GenericRecord>builder(new Path(file))
+            .withConf(conf)
+            .withSchema(schema)
+            .build();
+
+    //#init-writer
+
+    //#init-sink
+    Sink<GenericRecord, CompletionStage<Done>> sink = AvroParquetSink.create(writer);
+    //#init-sink
+
+    Source.from(records).runWith(sink, materializer);
+
+    Thread.sleep(1000);
+
+    assertEquals(records.size(), checkResponse());
+  }
+
+  private int checkResponse() throws IOException {
+
+    Path dataFile = new Path(file);
+    ParquetReader<GenericRecord> reader =
+        AvroParquetReader.<GenericRecord>builder(HadoopInputFile.fromPath(dataFile, conf))
+            .disableCompatibility()
+            .build();
+
+    List<GenericRecord> expectedRecords = Lists.newArrayList();
+    GenericRecord rec;
+    while ((rec = reader.read()) != null) {
+      expectedRecords.add(rec);
+    }
+    reader.close();
+    return expectedRecords.size();
+  }
+
+  @After
+  public void tearDown() {
+    system = null;
+    materializer = null;
+    File index = new File(folder);
+    String[] entries = index.list();
+    if (entries != null) {
+      for (String s : entries) {
+        File currentFile = new File(index.getPath(), s);
+        currentFile.delete();
```

Comment:
Did you consider to use `File.createTempFile`  and `File.deleteOnExit`?

Reply:
`File.createTempFile` can't work here, because I'm creating Hadoop Path file directly from path String, however `File.deleteOnExit` plays perfectly here. Thanks.